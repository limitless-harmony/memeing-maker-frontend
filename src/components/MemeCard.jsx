import React from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import { dark } from 'styles/colors';
import MemeText from 'components/MemeText';

const MemeCard = ({ src, square, small, topText, bottomText }) => {
  const defaultSrc = '/static/media/create-meme.bf0fa792.svg';
  return (
    <Meme square={square}>
      <MemeText square={square} text={topText} small={small} top />
      <svg
        style={{ display: 'block' }}
        width="100%"
        id="svg_ref"
        viewBox="0 0 400 400"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <image
          x="0"
          y="0"
          xlinkHref={src || defaultSrc}
          width="100%"
          height="100%"
        />
      </svg>
      <MemeText square={square} text={bottomText} small={small} />
    </Meme>
  );
};

const Meme = styled.div`
  margin: 0 0 ${calculateRem(11)};
  border-radius: ${({ square }) => (square ? 'none' : `${calculateRem(7)}`)};
  box-shadow: ${calculateRem(2)} ${calculateRem(4)} ${calculateRem(4)} 0 ${dark};
  width: 100%;
`;

export default MemeCard;
