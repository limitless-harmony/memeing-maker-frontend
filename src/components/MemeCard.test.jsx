import React from 'react';
import { shallow } from 'enzyme';
import MemeCard from 'components/MemeCard';

describe('Testing the MemeCard component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <MemeCard
        imageUrl="www.me.memememem"
        topText="A top text"
        bottomText="A bottom text"
      />
    );
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
