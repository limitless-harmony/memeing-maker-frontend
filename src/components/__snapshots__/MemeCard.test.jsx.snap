// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the MemeCard component renders without crashing 1`] = `
<styled.div>
  <MemeText
    text="A top text"
    top={true}
  />
  <svg
    id="svg_ref"
    style={
      Object {
        "display": "block",
      }
    }
    viewBox="0 0 400 400"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <image
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
  <MemeText
    text="A bottom text"
  />
</styled.div>
`;
